connector/src/main/scala/com/basho/riak/spark/rdd/RiakRDD.scala

```diff
@@ -65,6 +65,13 @@ class RiakRDD[R] private[spark](@transient sc: SparkContext,
     partitions
   }
 
+  override def getPreferredLocations(split: Partition) : Seq[String] = {
+    split match {
+      case (lcp: RiakLocalCoveragePartition[_]) => List(lcp.primaryHost.getHost)
```

Comment:
So if the `Partition` object passed in is a RiakLocalCoveragePartition, use the host provided inside of it.  Otherwise return Nil (the default - https://github.com/apache/spark/blob/master/core/src/main/scala/org/apache/spark/rdd/RDD.scala#L136)

Could we call the superclass instead of specifying `Nil`, or is `Nil` that widely known as the default value? :)